### PR TITLE
Generate barcode on asset creation

### LINKED ISSUE #625: [FEATURE] Automatically generate a barcode on asset creation
### Is there an existing issue or pull request for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

Asset location management relies on assets having barcodes associated with them. This functionality only exists in the rms app, which is deprecated.

### Describe the solution you'd like

Generate a basic asset barcode when a new asset is added. This should just be a generic value to be added to the assetBarcodes table that matches assets_tag

### Describe alternatives you've considered

Rewriting the whole asset location functionality to not have this dependency

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by cherry-john**
### Description

When an asset is created from an asset type, it will automatically create an associated barcode.

Closes #625 

### Checklist

- [x] I accept the contributor license agreement for this repository.
- [x] I have added documentation for new/changed functionality in this PR or in the [documentation repo](https://github.com/adam-rms/website).
- [x] I have updated the API documentation for any routes changed, within each api file.
- [x] All active GitHub checks for tests, formatting, and security are passing.
- [x] The correct base branch is being used (if not `main`).
- [x] A GitHub issue is linked to this pull request.

### CODE REVIEW

File: html/admin/api/assets/newAssetFromType.php [hunk 1]
```diff
@@ -27,7 +27,23 @@
 $result = $DBLIB->insert("assets", array_intersect_key( $array, array_flip( ['assets_tag','assetTypes_id','assets_notes','instances_id','asset_definableFields_1','asset_definableFields_2','asset_definableFields_3','asset_definableFields_4','asset_definableFields_5','asset_definableFields_6','asset_definableFields_7','asset_definableFields_8','asset_definableFields_9','asset_definableFields_10','assets_assetGroups'] ) ));
 
 if (!$result) finish(false, ["code" => "INSERT-FAIL", "message"=> "Could not insert asset"]);
-else finish(true, null, ["assets_id" => $result, "assets_tag" => $array['assets_tag'], "assetTypes_id" => $array['assetTypes_id']]);
+
+//Generate asset barcode
+
+$assetBarcodeData = [
+    "assetsBarcodes_value" => "A" . (is_numeric($asset['assets_tag']) ? $asset['assets_tag'] : $asset['assets_id']),
```
Comment (Jbithell) [inline html/admin/api/assets/newAssetFromType.php:34]: Is there a reason we can't just use the tag itself?

Comment (cherry-john) [reply html/admin/api/assets/newAssetFromType.php:34]: This follows how barcodes are generated elsewhere, happy to do this differently if you think that works best?

Comment (Jbithell) [reply html/admin/api/assets/newAssetFromType.php:34]: That might be a hangover from how tags used to work oops sorry - I reckon we could just use the tag, that way easier for those who have existing barcodes?

Comment (cherry-john) [reply html/admin/api/assets/newAssetFromType.php:34]: Sounds good 😆, should i update elsewhere too?

Comment (Jbithell) [reply html/admin/api/assets/newAssetFromType.php:34]: Sounds great, sorry 😢

**Review by Jbithell — APPROVED** [review]

**Pull request merged**